jax/_src/numpy/lax_numpy.py

```diff
@@ -1777,13 +1777,13 @@ def _where(condition, x=None, y=None):
 successfully. Alternatively, you can specify the optional ``size`` keyword:
 if specified, the first ``size`` True elements will be returned; if there
 are fewer True elements than ``size`` indicates, the index arrays will be
-padded with zeros.
+padded with ``fill_value`` (default is 0.)
 """
 
 @_wraps(np.where, update_doc=False, lax_description=_WHERE_DOC)
-def where(condition, x=None, y=None, *, size=None):
+def where(condition, x=None, y=None, *, size=None, fill_value=None):
   if x is None and y is None:
-    return nonzero(asarray(condition), size=size)
+    return nonzero(asarray(condition), size=size, fill_value=fill_value)
   else:
     if size is not None:
```

Comment:
Perhaps check if `fill_value is not None` here as well?

Reply:
How's this? The semantics are the same but it only passes down `fill_value` if provided.

Reply:
Sorry to be unclear: I meant below, where it raises an error `if size is not None`. We should change that to `if size is not None or fill_value is not None` and change the error text appropriately.

I think it's fine to pass `fill_value` to `nonzero` regardless of its value.

Reply:
Got it!